scarlet/source.py

```diff
@@ -1,396 +1,338 @@
-                              config=config)
-        component._normalize(normalization)
-        super(ExtendedSource, self).__init__(component)
+class ExtendedSource(PointSource):
+    """Extended source intialized to match a set of observations
 
-    def _make_initial(self, center, img, bg_rms, thresh=1., symmetric=True, monotonic=True, config=None):
-        """Initialize the source that is symmetric and monotonic
+    Parameters
+    ----------
+    sky_coord: tuple
+        Center of the source
+    scene: `scarlet.observation.Scene`
+        The scene that the model lives in.
+    observations: list of `~scarlet.observation.Observation`
+        Observations to extract SED from.
+    bg_rms: array
+        Background RMS in each band in observation.
+    obs_idx: int
+        Index of the observation in `observations` to use for
+        initializing the morphology.
+    thresh: `float`
+        Multiple of the backround RMS used as a
+        flux cutoff for morphology initialization.
+    symmetric: `bool`
+        Whether or not to enforce symmetry.
+    monotonic: `bool`
+        Whether or not to make the object monotonically decrease
+        in flux from the center.
+    component_kwargs: dict
+        Keyword arguments to pass to the component initialization.
+    """
+    def __init__(self, sky_coord, scene, observations, bg_rms, obs_idx=0, thresh=1,
+                 symmetric=False, monotonic=True, center_step=5, delay_thresh=0, **component_kwargs):
+        self.symmetric = symmetric
+        self.monotonic = monotonic
+        self.coords = sky_coord
+        center = scene.get_pixel(sky_coord)
+        self.pixel_center = center
+        self.center_step = center_step
+        self.delay_thresh = delay_thresh
 
-        See `self.__init__` for a description of the parameters
-        """
-        # determine initial SED from peak position
-        B = img.shape[0]
-        center_int = np.round(center).astype('int')
-        sed = get_pixel_sed(img, center_int) # amplitude is in sed
-
-        # build optimal detection coadd given the sed
-        if np.all([bg_rms > 0]):
-            bg_rms = np.array(bg_rms)
-            weights = np.array([sed[b]/bg_rms[b]**2 for b in range(B)])
-            jacobian = np.array([sed[b]**2/bg_rms[b]**2 for b in range(B)]).sum()
-            detect = np.einsum('i,i...', weights, img) / jacobian
-
-            # thresh is multiple above the rms of detect (weighted variance across bands)
-            bg_cutoff = thresh * np.sqrt((weights**2 * bg_rms**2).sum()) / jacobian
-        else:
-            detect = np.sum(img, axis=0)
-            bg_cutoff = 0
-        morph = self._init_morph(detect, center, bg_cutoff, symmetric, monotonic, config)
+        sed, morph = init_extended_source(sky_coord, scene, observations, bg_rms, obs_idx,
+                                          thresh, True, monotonic)
```

Comment:
It's a bit weird that the initialization is always symmetric even if the source itself may not have to. I think this isn't transparent to the user.

Reply:
I get that, but this really helps initialize compact sources and makes a big difference in the convergence time. Since we expect users looking at more complicated objects to override the initialization anyway, I'm fine with hiding this under the hood.